src/jit/lsraarmarch.cpp

```diff
@@ -552,74 +552,69 @@ int LinearScan::BuildPutArgSplit(GenTreePutArgSplit* argNode)
 #endif // FEATURE_ARG_SPLIT
 
 //------------------------------------------------------------------------
-// BuildBlockStore: Set the NodeInfo for a block store.
+// BuildBlockStore: Build the RefPositions for a block store node.
 //
 // Arguments:
-//    blkNode       - The block store node of interest
+//    blkNode - The block store node of interest
 //
 // Return Value:
 //    The number of sources consumed by this node.
 //
 int LinearScan::BuildBlockStore(GenTreeBlk* blkNode)
 {
-    GenTree* dstAddr  = blkNode->Addr();
-    unsigned size     = blkNode->Size();
-    GenTree* source   = blkNode->Data();
-    int      srcCount = 0;
+    GenTree* dstAddr = blkNode->Addr();
+    GenTree* src     = blkNode->Data();
+    unsigned size    = blkNode->Size();
 
     GenTree* srcAddrOrFill = nullptr;
-    bool     isInitBlk     = blkNode->OperIsInitBlkOp();
 
-    regMaskTP dstAddrRegMask        = RBM_NONE;
-    regMaskTP sourceRegMask         = RBM_NONE;
-    regMaskTP blkSizeRegMask        = RBM_NONE;
-    regMaskTP internalIntCandidates = RBM_NONE;
+    regMaskTP dstAddrRegMask = RBM_NONE;
+    regMaskTP srcRegMask     = RBM_NONE;
+    regMaskTP sizeRegMask    = RBM_NONE;
 
-    if (isInitBlk)
+    if (blkNode->OperIsInitBlkOp())
     {
-        GenTree* initVal = source;
-        if (initVal->OperIsInitVal())
+        if (src->OperIs(GT_INIT_VAL))
         {
-            assert(initVal->isContained());
-            initVal = initVal->gtGetOp1();
+            assert(src->isContained());
+            src = src->AsUnOp()->gtGetOp1();
         }
-        srcAddrOrFill = initVal;
 
-        if (blkNode->gtBlkOpKind == GenTreeBlk::BlkOpKindUnroll)
-        {
-            // TODO-ARM-CQ: Currently we generate a helper call for every
```

Comment:
Is there a reason you removed this `TODO`?

Reply:
The NYI_ARM bellow should suffice. And anyway I'll enable unroll on ARM in the "improve" PR: https://github.com/dotnet/coreclr/pull/21711/commits/d871e371ff2d66a571a628c340d6b1d244e8e751

Reply:
OK, I can live with that, though believe it or not, I do occasionally grep for the TODOs.

Reply:
We have that too but in lowering: https://github.com/dotnet/coreclr/blob/2424f348fc7f23a35752fc078e7bbdcbfad96ad1/src/jit/lowerarmarch.cpp#L280-L285